### ISSUE TITLE
[Style/SoleNestedConditional] Syntax error when autocorrecting nested if conditions with a block

### ISSUE THREAD

**Issue by tom-kuca** [rubocop/rubocop#10881]
If there are nested `if` statesments, where the second `if` condition is a function with an arugment and a block, autocorrecting causes a syntax error. Given snippet `test.rb`

```ruby
def foo
  if cond
    if validate :strict do
        "hello"
      end
    end
  end
end
```

and running 

```
rubocop test.rb --force-default-config --only Style/SoleNestedConditional -a
```

there is a syntax error reported by `ruby` (see below). Prior to autocorrect, there was no syntax error.

--------

## Expected behavior

`ruby test.rb` do not report syntax error after autocorrect. Proposed solution is to add parenthesis around arguments.

```ruby
def foo
  if cond && validate(:strict) do
        "hello"
      end
    end
end
```

## Actual behavior

The source code is autocorrected to

```ruby
def foo
  if cond && validate :strict do
        puts "hello"
      end
    end
end
```

`ruby test.rb` reports syntax error:

```
/tmp/test.rb:2: syntax error, unexpected symbol literal, expecting `do' or '{' or '('
  if cond && validate :strict do
/tmp/test.rb:5: syntax error, unexpected `end', expecting end-of-input
    end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.33.0 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 2.7.4 x86_64-linux)
```

**Comment by tom-kuca** [comment]
Thanks for the fix!
